File: Task/Look-and-say-sequence/Nim/look-and-say-sequence.nim
iterator lookAndSay(n: int): string =
  var current = "1"
  yield current
  for round in 2..n:
    var ch = current[0]
    var count = 1
    var next = ""
    for i in 1..current.high:
      if current[i] == ch:
        inc count
      else:
        next.add $count & ch
        ch = current[i]
        count = 1
    current = next & $count & ch
    yield current

for s in lookAndSay(12):
  echo s

File: Task/Look-and-say-sequence/PascalABC.NET/look-and-say-sequence.pas
##
function lookAndSay(): sequence of string;
begin
  var current: string := '1';
  yield current;

  while true do
  begin
    var ch := current[1];
    var count := 1;
    var next := '';
    foreach var i in 2..current.Length do
      if current[i] = ch then
        inc(count)
      else begin
        next += count.ToString + ch;
        ch := current[i];
        count := 1;
      end;
    current := next + count.ToString + ch;
    yield current;
  end;
end;

foreach var s in lookandsay.Take(12) do
  s.Println;
